hawc/apps/riskofbias/serializers.py

```diff
@@ -125,18 +130,127 @@ class Meta:
+                for rob in robs:
+                    if rob.active and rob.final:
+                        raise serializers.ValidationError(
+                            "create failed; study %s already has an active & final risk of bias"
+                            % study_id
+                        )
+
+            # (2) subject to the settings of the metrics defined for this assessment,
+            # and the study_type of the study, we need to make sure all necessary scores
+            # have been submitted with a default=True
+            scores = self.initial_data["scores"]
+            required_metrics = models.RiskOfBiasMetric.objects.get_required_metrics(
+                assessment, study
+            )
+            problematic_scores = []
+            for metric in required_metrics:
+                domain = metric.domain
+                # there could be multiple scores for a given metric if there are overrides, so we need to fetch them all
+                scores_for_metric = [
+                    score
+                    for score in scores
+                    if "metric_id" in score and score["metric_id"] == metric.id
+                ]
+
+                metric_descriptor = "'%s:%s'" % (domain.name, metric.name)
+
+                if 0 == len(scores_for_metric):
+                    problematic_scores.append(
+                        "No score for metric %s/%s was submitted" % (metric.id, metric_descriptor)
+                    )
+                else:
+                    default_score_for_this_metric_included = False
+                    for submitted_score in scores_for_metric:
+                        if (
+                            "is_default" in submitted_score
+                            and submitted_score["is_default"] is True
+                        ):
+                            default_score_for_this_metric_included = True
+                            break
+
+                    if not default_score_for_this_metric_included:
+                        problematic_scores.append(
+                            "The score for metric %s/%s was submitted but not marked as default"
+                            % (metric.id, metric_descriptor)
+                        )
+
+            if len(problematic_scores) > 0:
+                explanation = "; ".join(problematic_scores)
+                raise serializers.ValidationError(
+                    "create failed; study %s had problematic scores (%s)" % (study_id, explanation)
+                )
+
+            # store the actual metric object we want to create
+            score_idx = 0
+            for score in data["scores"]:
+                metric = models.RiskOfBiasMetric.objects.get(id=scores[score_idx]["metric_id"])
+                # score["metric_id"] = scores[score_idx]["metric_id"]
+                score["metric"] = metric
+                score_idx += 1
+
+            override_options = models.RiskOfBias().get_override_options()
```

Comment:
need to specify `study=data['study']` for this to return something i think

Reply:
@shapiromatron not sure what you mean by "for this to return something" - like do you want more than just the study_id (title, abstract, etc.) included in the JSON that goes back to whoever called the API?

Reply:
ohh I think I remember now. You instantiate a new `RiskOfBias` object which is unbound to a study, so the options are querying based on null lookups. If you could attach it to an existing rob instance that would be preferred, does one exist that you could use?

Reply:
Ah got it - ok, fixed that, fixed the check against override_options to actually be used when creating, and added some tests for good/bad override options during a create API call.